langlib/lang.runtime/src/main/ballerina/runtime.bal

```diff
@@ -84,10 +84,15 @@ public isolated function getStackTrace() returns StackFrame[] {
     int i = 0;
     CallStackElement[] callStackElements = externGetStackTrace();
     lang_array:forEach(callStackElements, function (CallStackElement callStackElement) {
-                                stackFrame[i] = new java:StackFrameImpl(callStackElement.callableName,
-                                callStackElement.moduleName, callStackElement.fileName, callStackElement.lineNumber);
-                                i += 1;
-                            });
+        if (callStackElement["moduleName"] is string) {
+            stackFrame[i] = new java:StackFrameImpl(callStackElement.callableName, callStackElement.moduleName,
+            callStackElement.fileName, callStackElement.lineNumber);
+        } else {
```

Comment:
Do we ever get to the else block? `CallStackElement` is still

```ballerina
type CallStackElement record {|
    string callableName;
    string moduleName;
    string fileName;
    int lineNumber;
|};
```

Shouldn't `moduleName` be an optional field there?

Reply:
Yup, for single bal file case we were able to reach the else block. Refer the unit tests added for this in the commit https://github.com/ballerina-platform/ballerina-lang/pull/30998/commits/88be45ed8ce185ece1b6376e8cd6dffb877a91ab. (Anyway this if else check has been removed with the new changes addressed with the commit https://github.com/ballerina-platform/ballerina-lang/pull/30998/commits/e88161f75a7c26f69317defdd98388059a53ab5d)

Reply:
Yes, this will now get hit. But previously, when `moduleName` was not an optional field, it wouldn't have. At least not according to b7a semantics (unless the value was invalidly set in native code).